src/cryptoadvance/specter/node.py

```diff
@@ -178,44 +189,57 @@ def update_rpc(
         host=None,
         protocol=None,
     ):
+        """Changes the attributes of that node but only persists it, if the rpc.test_connection succeeds"""
         update_rpc = self.rpc is None or not self.rpc.test_connection()
         if autodetect is not None and self.autodetect != autodetect:
+            logger.debug(f"{self} updating autodetect to {autodetect}")
             self.autodetect = autodetect
             update_rpc = True
         if datadir is not None and self.datadir != datadir:
+            logger.debug(f"{self} updating datadir to {datadir}")
             self.datadir = datadir
             update_rpc = True
         if user is not None and self.user != user:
+            logger.debug(f"{self} updating user to {user}")
             self.user = user
             update_rpc = True
         if password is not None and self.password != password:
+            logger.debug(f"{self} updating password to XXXXXXXX")
             self.password = password
             update_rpc = True
         if port is not None and self.port != port:
+            logger.debug(f"{self} updating port to {port}")
             self.port = port
             update_rpc = True
         if host is not None and self.host != host:
+            logger.debug(f"{self} updating host to {host}")
             self.host = host
             update_rpc = True
         if protocol is not None and self.protocol != protocol:
+            logger.debug(f"{self} updating protocol to {protocol}")
             self.protocol = protocol
             update_rpc = True
         if update_rpc:
             self.rpc = self.get_rpc()
-            write_node(self, self.fullpath)
+            if self.rpc and self.rpc.test_connection():
+                logger.info(f"persisting {self} in update_rpc")
+                write_node(self, self.fullpath)
+            else:
+                logger.error(f"not persisting broken {self.rpc} in update_rpc")
```

Comment:
I think this fixed the flaky test. `update_rpc()` is called way too often but the circumstances it'll change the persisted node was not specified good enough.